runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/PortableBundleManager.java

```diff
@@ -110,15 +112,22 @@ private void scheduleNextBundleCheck() {
   @Override
   public void tryStartBundle() {
 
-    currentBundleElementCount++;
+    currentBundleElementCount.incrementAndGet();
```

Comment:
Is it recoverable if `onBundleStarted` failed?

If we go to `signalFailure`, it might not matter?

Reply:
I think is is a readability issue.

If `onBundleStarted` fails, `DoFnOp` catches the exception, and invokes `singalFailure`, the state is then reset.

[See code here](https://github.com/apache/beam/blob/master/runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/DoFnOp.java#L305)

Reply:
On the failure case moving to the end would change from
"increment then reset" to only "reset"

It might reduce the number of race conditions possible, since it is 1 less state change.